priiloader/source/hacks.cpp

```diff
@@ -549,14 +563,62 @@ s8 LoadSystemHacks(bool load_nand)
 
 			mem_free(fbuf);
 
-			//Set all system_hacks for system menu > max_version || sysver < min_version to 0
+			
 			unsigned int sysver = GetSysMenuVersion();
+			
+			// Make sure that all master hacks are disabled in the state array. 
+			// That way, when all sub-hacks are disabled, the master hack will 
+			// be disabled as well. Then, in the next loop, if a sub hack is 
+			// selected, the master hack will be enabled again. 
+			for( u32 i=0; i < system_hacks.size(); ++i)
+			{
+				if (system_hacks[i].provide.length() > 0) {
+					// This is a master hack
+					states_hash[i] = 0; 
+				}
+			}
+			
+			
+			//Set all system_hacks for system menu > max_version || sysver < min_version to 0
 			for( u32 i=0; i < system_hacks.size(); ++i)
 			{
 				if( system_hacks[i].max_version < sysver || system_hacks[i].min_version > sysver)
 				{
 					states_hash[i] = 0;
 				}
+				
+				// If there is a hack that requires another one, make sure that that is present: 
+				else if (system_hacks[i].require.length() > 0 )  {

```

Comment:
Why is this the wrong place? I'm just checking, for each sub hack, if the needed master hack exists. If it doesn't exist, I force disable the hack because sub hacks without their master hack don't work. 
And yeah, I am looping three times there - why not? First disabling all master hacks, then checking the hack requirements. It's just a simple for loop with a simple condition - it's not like that is going to take ages. 

Would you know a better place to check all that? 
I could remove all the checks if you wanted to, but I thought it would be better to have a failsafe and disable broken hacks, rather than letting the user use corrupted hacks ...